### ISSUE TITLE
Some python `whl` files are detected as Chef packages

### ISSUE THREAD

**Issue by majurg** [nexB/scancode-toolkit#1600]
![Screenshot from 2019-06-13 12-40-22](https://user-images.githubusercontent.com/7485204/59462572-dd7e3900-8dd8-11e9-918a-09026fefd627.png)

Using the latest `scancode-toolkit` develop checkout when scanning an extracted python `.whl` file, it is recognized as a Chef package. 

From the above image, you can see that the package is being created at the `abooutcode_toolkit-3.2.1.dist-info/` level, and the data is gathered from the `metadata.json` file. This metadata file is correct on the whole for `name`, `version` and `declared_license` fields, but it is catagorized as a Chef and Ruby package with wrong download URLs. 

@JonoYang I believe you worked on adding Chef packages? Not sure if we were detected python packages in this way before, so your additions are useful for these python `whl` files, just needs some tweaking on both the python and chef package ends to properly detect both. 

Let me know if you need anything else from me.

**Comment by majurg** [comment]
Oh and not every python `.whl` is detected like this. Only when it has a `metadata.json` file in its `*.dist-info` directory/

**Comment by JonoYang** [comment]
@majurg This bug is happening because Chef metadata files are also named `metadata.json`. A stopgap fix I can think of is to also put in a check for the directory name or the name of some other file next to `metadata.json` that would give it an idea if it is inside a whl's dist-info directory

**Comment by majurg** [comment]
@JonoYang Do Chef packages contain a parent `*.dist-info` directory? If not, this can probably be a good proxy as well.

**Comment by JonoYang** [comment]
@majurg No. Chef `metadata.json`'s seem to always be at the root of the Chef recipe it's for. I'll make the change.

**Comment by JonoYang** [comment]
@majurg I've made an update to the metadata.json detection check. Let me know if it's still acting up.

**Comment by majurg** [comment]
@JonoYang I will take a look shortly.
